prov/rstream/src/rstream_init.c

```diff
@@ -86,9 +92,36 @@ static int rstream_getinfo(uint32_t version, const char *node,
 		rstream_info.domain_attr->max_ep_srx_ctx = 0;
 	}
 
+	/* Avoid getting wild card address from MSG provider */
+	if (ofi_is_wildcard_listen_addr(node, service, flags, hints)) {
+		if (service) {
+			ret = getaddrinfo(NULL, service, NULL, &ai);
+			if (ret) {
+				FI_WARN(&rstream_prov, FI_LOG_CORE,
+					"Unable to getaddrinfo\n");
+				return ret;
+			}
+			port_save = ofi_addr_get_port(ai->ai_addr);
+			freeaddrinfo(ai);
+			service = NULL;
+		} else {
+			if (hints) {
+				port_save = ofi_addr_get_port(hints->src_addr);
+				ofi_addr_set_port(hints->src_addr, 0);
```

Comment:
Hmm.. actually hints is a const here.  So it cannot change. We can't modify the user's structure.